x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/hooks/metrics_to_api_options.ts

```diff
@@ -57,27 +67,43 @@ export interface NodeMetricsExplorerOptionsMetric<Field extends string>
   field: Field;
 }
 
-export function metricsToApiOptions<T extends string>(metricsMap: MetricsMap<T>, groupBy: string) {
-  const metrics = Object.values(metricsMap) as Array<NodeMetricsExplorerOptionsMetric<T>>;
+export function metricsToApiOptions<T extends string>(
+  metricsQueryOptions: MetricsQueryOptions<T>,
+  filterClauseDsl?: QueryDslQueryContainer
+) {
+  const metrics = Object.values(metricsQueryOptions.metricsMap) as Array<
+    NodeMetricsExplorerOptionsMetric<T>
+  >;
 
   const options: MetricsExplorerOptions = {
     aggregation: 'avg',
-    groupBy,
+    groupBy: metricsQueryOptions.groupByField,
     metrics,
+    filterQuery: JSON.stringify(
+      buildFilterClause(metricsQueryOptions.sourceFilter, filterClauseDsl)
+    ),
   };
 
-  const metricByField = createFieldLookup(Object.keys(metricsMap) as T[], metrics);
-
   return {
     options,
-    metricByField,
   };
 }
 
-function createFieldLookup<T extends string>(
-  fields: T[],
-  metrics: Array<NodeMetricsExplorerOptionsMetric<T>>
-) {
+function buildFilterClause(
+  sourceFilter: QueryDslQueryContainer,
+  filterClauseDsl?: QueryDslQueryContainer
+): QueryDslQueryContainer {
+  return {
+    bool: {
+      filter: !!filterClauseDsl ? [sourceFilter, filterClauseDsl] : [sourceFilter],
+    },
+  };
+}
+
+export function createMetricByFieldLookup<T extends string>(metricMap: MetricsMap<T>) {
+  const fields = Object.keys(metricMap) as T[];
+  const metrics = Object.values(metricMap) as Array<NodeMetricsExplorerOptionsMetric<T>>;
```

Comment:
Nit: These assertions could be even safer if we let the compiler derive the types, for example:

```suggestion
  const fields = Object.keys(metricMap) as Array<keyof typeof metricMap>;
  const metrics = Object.values(metricMap) as ObjectValues<typeof metricMap>;
```

where `ObjectValues<T>` is defined in https://github.com/elastic/kibana/blob/1c08fb4cfb8c5365d5eb3ae5de58ea71b0ed07fd/x-pack/plugins/infra/common/utility_types.ts#L47

What do you think?

Reply:
nice! seems neat